src/main/java/ru/yandex/clickhouse/BalancedClickhouseDataSource.java

```diff
@@ -0,0 +1,234 @@
+       }
+        for (final String url : urls) {
+            log.debug("Pinging enabled url: " + url);
+            if (!ping(url)) {
+                log.debug("Url is dead now: " + url);
+                urlsToDisable.add(url);
+            }
+        }
+
+        for (final String url: urlsToEnable) {
+            addUrl(url);
+        }
+
+        for (final String url: urlsToDisable) {
+            disableUrl(url);
+        }
+    }
+
+
+    private String getAnyUrl() {
+        final List<String> sources = new ArrayList<String>(urls);
+        if (sources.isEmpty()) {
+            throw new RuntimeException("Unable to get connection: there is no enabled urls");
+        }
+        final int idx = rnd.nextInt(sources.size());
+        return sources.get(idx);
+    }
+
+    @Override
+    public Connection getConnection() throws SQLException {
+        return driver.connect(getAnyUrl(), properties);
+    }
+
+    @Override
+    public Connection getConnection(String username, String password) throws SQLException {
+        return driver.connect(getAnyUrl(), properties.withCredentials(username, password));
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> iface) throws SQLException {
+        if (iface.isAssignableFrom(getClass())) {
+            return iface.cast(this);
+        }
+        throw new SQLException("Cannot unwrap to " + iface.getName());
+    }
+
+    @Override
+    public boolean isWrapperFor(Class<?> iface) throws SQLException {
+        return iface.isAssignableFrom(getClass());
+    }
+
+    @Override
+    public PrintWriter getLogWriter() throws SQLException {
+        return printWriter;
+    }
+
+    @Override
+    public void setLogWriter(PrintWriter printWriter) throws SQLException {
+        this.printWriter = printWriter;
+    }
+
+    @Override
+    public void setLoginTimeout(int seconds) throws SQLException {
+        loginTimeout = seconds;
+    }
+
+    @Override
+    public int getLoginTimeout() throws SQLException {
+        return loginTimeout;
+    }
+
+    public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+        throw new SQLFeatureNotSupportedException();
+    }
+
+    public BalancedClickhouseDataSource withConnectionsCleaning(int rate, TimeUnit timeUnit) {
+        driver.scheduleConnectionsCleaning(rate, timeUnit);
+        return this;
+    }
+
+    public void scheduleActualization(int rate, TimeUnit timeUnit){
+        BalancedClickhouseDataSource.ScheduledActualizer.INSTANCE.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    actualize();
+                } catch (Exception e){
+                    log.error("Unable to actualize urls: " + e);
+                }
+            }
+        }, 0, rate, timeUnit);
+    }
+
+    private static class ScheduledActualizer {
+        private static final ScheduledExecutorService INSTANCE = Executors.newSingleThreadScheduledExecutor();
+    }
```

Comment:
It means, I have to do it "public" in a ClickhouseDriver. If it is acceptable, ok...